src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/MethodConfigurationComponent.scala

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.dsde.rawls.dataaccess.slick
+
+case class MethodConfigurationRecord(id: Long,
+                                     namespace: String,
+                                     name: String,
+                                     rootEntityType: String,
+                                     methodNamespace: String,
+                                     methodName: String,
+                                     methodVersion: Int)
+
+case class MethodConfigurationInputRecord(methodConfigId: Long, id: Long, key: String, value: String)
+
+case class MethodConfigurationOutputRecord(methodConfigId: Long, id: Long, key: String, value: String)
+
+case class MethodConfigurationPrereqRecord(methodConfigId: Long, id: Long, key: String, value: String)
+
+trait MethodConfigurationComponent {
+  this: DriverComponent =>
+
+  import driver.api._
+
+  class MethodConfigurationTable(tag: Tag) extends Table[MethodConfigurationRecord](tag, "METHOD_CONFIG") {
+    def id = column[Long]("ID", O.PrimaryKey, O.AutoInc)
+    def namespace = column[String]("NAMESPACE")
+    def name = column[String]("NAME")
+    def rootEntityType = column[String]("ROOT_ENTITY_TYPE")
+    def methodNamespace = column[String]("METHOD_NAMESPACE")
+    def methodName = column[String]("METHOD_NAME")
+    def methodVersion = column[Int]("METHOD_VERSION")
+
+    def * = (id, namespace, name, rootEntityType, methodNamespace, methodName, methodVersion) <> (MethodConfigurationRecord.tupled, MethodConfigurationRecord.unapply)
+
+    def namespaceNameIdx = index("NAMESPACE_NAME_IDX", (namespace, name), unique = true)
```

Comment:
Needs some kind of method config prefix because indices need to be globally unique